core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
   return(explainTuple);
 }
 
+static void appendListOfColumns(Queue* listOfColNames,ComTdb *tdb, NAString& outNAString){
+
+	if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      Lng32 currPos = 0;
+	      Lng32 jj = 0;
+	      short colNameLen = *(short*)currPtr;
+	      currPos += sizeof(short);
+	      char colFam[100];
```

Comment:
as stated above this is a function crated out of the model in ComTdbHbaseAccess.cpp. So any refactoring / comment suggestion would be on the JIRA to refactor/fix the buffer overrun in a consistant manner